stages/org.osbuild.ovf

```diff
@@ -0,0 +1,152 @@
+        <rasd:Description>Number of Virtual CPUs</rasd:Description>
+        <rasd:ElementName>2 virtual CPU(s)</rasd:ElementName>
+        <rasd:InstanceID>1</rasd:InstanceID>
+        <rasd:ResourceType>3</rasd:ResourceType>
+	<rasd:VirtualQuantity>2</rasd:VirtualQuantity>
+      </Item>
+      <Item>
+        <rasd:AllocationUnits>byte * 2^20</rasd:AllocationUnits>
+        <rasd:Description>Memory Size</rasd:Description>
+        <rasd:ElementName>4096 MB of memory</rasd:ElementName>
+        <rasd:InstanceID>2</rasd:InstanceID>
+        <rasd:ResourceType>4</rasd:ResourceType>
+	<rasd:VirtualQuantity>4096</rasd:VirtualQuantity>
+      </Item>
+      <Item ovf:required="false">
+        <rasd:AutomaticAllocation>false</rasd:AutomaticAllocation>
+        <rasd:ElementName>VirtualVMCIDevice</rasd:ElementName>
+        <rasd:InstanceID>7</rasd:InstanceID>
+        <rasd:ResourceSubType>vmware.vmci</rasd:ResourceSubType>
+        <rasd:ResourceType>1</rasd:ResourceType>
+        <vmw:Config ovf:required="false" vmw:key="allowUnrestrictedCommunication" vmw:value="false"/>
+      </Item>
+      <Item>
+        <rasd:Address>0</rasd:Address>
+        <rasd:Description>SCSI Controller</rasd:Description>
+        <rasd:ElementName>SCSI Controller 0</rasd:ElementName>
+        <rasd:InstanceID>3</rasd:InstanceID>
+	<rasd:ResourceSubType>VirtualSCSI</rasd:ResourceSubType>
+        <rasd:ResourceType>6</rasd:ResourceType>
+      </Item>
+      <Item>
+        <rasd:Address>0</rasd:Address>
+        <rasd:Description>IDE Controller</rasd:Description>
+        <rasd:ElementName>VirtualIDEController 0</rasd:ElementName>
+        <rasd:InstanceID>4</rasd:InstanceID>
+        <rasd:ResourceType>5</rasd:ResourceType>
+      </Item>
+      <Item>
+        <rasd:AddressOnParent>0</rasd:AddressOnParent>
+        <rasd:ElementName>Hard disk 0</rasd:ElementName>
+        <rasd:HostResource>ovf:/disk/vmdisk1</rasd:HostResource>
+        <rasd:InstanceID>5</rasd:InstanceID>
+        <rasd:Parent>3</rasd:Parent>
+        <rasd:ResourceType>17</rasd:ResourceType>
+        <vmw:Config ovf:required="false" vmw:key="backing.writeThrough" vmw:value="false"/>
+      </Item>
+      <vmw:Config ovf:required="false" vmw:key="bootOptions.efiSecureBootEnabled" vmw:value="false"/>
+      <vmw:Config ovf:required="false" vmw:key="firmware" vmw:value="efi"/>
+    </VirtualHardwareSection>
+  </VirtualSystem>
+</Envelope>
+"""
+
+
+def virtual_size(vmdk):
+    cmd = ["qemu-img", "info", "--output=json", vmdk]
+    res = subprocess.run(cmd, check=True, capture_output=True)
+    if res.returncode != 0:
+        raise RuntimeError("Unable to determine vmdk size")
+    return json.loads(res.stdout)["virtual-size"]
+
+
+def write_template(vmdk):
+    basename = os.path.basename(vmdk)
+    ovf_data = OVF_TEMPLATE.format(vmdk_size=os.stat(vmdk).st_size,
+                                   vmdk_capacity=virtual_size(vmdk), image_name=basename)
```

Comment:
So the image name just needs to refer to the basename of the disk name. When importing the ova I assume that vSphere untars it, and then looks for the disk based on the image_name in the ovf.

Reply:
Right. What I'm not sure about is if would be a problem for vSphere to have multiple ova's imported and both of them referencing the same `image_name` in the ovf. If vSphere stores the content of the ova in a different path, then it should not be an issue. Also I was not sure what will vSphere display as the image name (if that is even a thing when you import ova).

Reply:
So I just tried to import the same OVA twice, and it does complain  that a VM of the same name exists. That said you can just edit the name right there. This also makes sense from the POV of VM-oriented (rather than pure disk) "images".

![image](https://user-images.githubusercontent.com/11140201/227512255-8feac9ef-23a8-4966-a05b-2afcdae7a12e.png)

![image](https://user-images.githubusercontent.com/11140201/227512601-7ce67faa-8a43-44b4-9cdf-18f8e1df6f8a.png)